huff0/decompress.go

```diff
@@ -641,90 +622,6 @@ func (d *Decoder) decompress1X8Bit(dst, src []byte) ([]byte, error) {
 	return dst, br.close()
 }
 
-// decompress1X8Bit will decompress a 1X encoded stream with tablelog <= 8.
-// The cap of the output buffer will be the maximum decompressed size.
-// The length of the supplied input must match the end of a block exactly.
-func (d *Decoder) decompress1X8BitExactly(dst, src []byte) ([]byte, error) {
-	var br bitReaderBytes
-	err := br.init(src)
-	if err != nil {
-		return dst, err
-	}
-	maxDecodedSize := cap(dst)
-	dst = dst[:0]
-
-	// Avoid bounds check by always having full sized table.
-	dt := d.dt.single[:256]
-
-	// Use temp table to avoid bound checks/append penalty.
-	bufs := d.buffer()
-	buf := &bufs[0]
-	var off uint8
-
-	const shift = 56
-
-	//fmt.Printf("mask: %b, tl:%d\n", mask, d.actualTableLog)
-	for br.off >= 4 {
-		br.fillFast()
-		v := dt[uint8(br.value>>shift)]
```

Comment:
Main difference is this being a constant shift in this mode and `br.peekByteFast()` being used below.

I would expect this to be faster in Go.

Reply:
Right, I'll benchmark this and get back to that with some results.

Reply:
@klauspost you were right, there are significant regressions